sdk/src/Core/Amazon.Runtime/Internal/InternalConfiguration.cs

```diff
@@ -182,9 +182,17 @@ private bool TryGetEnvironmentVariable(string environmentVariableName, out strin
                 {
                     convertedValue = int.Parse(value);
                 }
+                else if (typeof(T) == typeof(long))
```

Comment:
These changes were bug fixes after I change this code in previous PR to longer use the [TypeDescriptor](https://github.com/aws/aws-sdk-net/blob/main/sdk/src/Core/Amazon.Runtime/Internal/InternalConfiguration.cs#L177C29-L177C43) because that wasn't compatible for trimming. I looked where these private method was being used and confirmed these are the only types used for the generic `T`. Actually the `string` is not used but I added it for completeness.